crates/matrix-sdk-store-encryption/Cargo.toml

```
@@ -9,8 +9,10 @@ rust-version = "1.60"
 
 [features]
 js = ["getrandom/js"]
+default = ["bincode"]
```

so, wait. this would include both bincode and json_serde then?

Also, if we changed this default, we'd have a breaking change here - as it won't be able deserialise value written before with serde_json. I can see two possible ways out:
1. make bincode non-default and disable serde_json when not in use (which is what I'd prefer for now) on this crate
2. provide fallback support, to read old values with serde_json, too - but only encrypt on save with the new one. Not sure how feasible this is though...
Either way, for now:

```suggestion
default = []
bincode = ["dep:bincode"]
```

---

the deserialize helper function attempts to read with serde_json first, then bincode. I added a test that makes sure that old ciphertexts could still be decoded (`decrypt_json`)